DailyBudget/DailyBudgetWidget/DailyBudgetWidget.swift

```diff
@@ -0,0 +1,204 @@
+    guard
+      let id = entry.entity?.id,
+      let budget = budgets.first(where: { $0.uuid == id })
+    else { return nil }
+    
+    return BudgetProgressInfo(budget: budget, date: .today)
+  }
+  
+  var body: some View {
+    if let info {
+      VStack(alignment: .center) {
+        AmountText(
+          amount: info.currentAllowance,
+          wholePartFont: .system(
+            size: UIFont.preferredFont(
+              forTextStyle: .largeTitle).pointSize * 2)
+        )
+        .minimumScaleFactor(0.1)
+        .bold()
+        .foregroundStyle(.green)
+        
+        Text("Available today")
+          .fontWeight(.light)
+        
+        Text(info.budget.name)
+          .font(.subheadline)
+          .bold()
+          .lineLimit(2)
+          .multilineTextAlignment(.center)
+      }
+      
+    } else {
+      Text("No budget selected").foregroundStyle(.gray)
+      Text("Long-press to configure")
+        .font(.footnote)
+    }
+  }
+}
+
+// TODO: Include
+//struct DailyBudgetWidgetView2: View {
+//  var entry: BudgetEntry
+//  
+//  var body: some View {
+//    VStack {
+//      HStack {
+//        Text(entry.title)
+//          .font(.subheadline)
+//          .bold()
+//          .lineLimit(2)
+//          .multilineTextAlignment(.center)
+//        Spacer()
+//        Image(systemName: "calendar")
+//        Text(entry.dateText)
+//      }
+//      .font(.footnote)
+//      
+//      Spacer()
+//      
+//      Grid {
+//        GridRow(alignment: .lastTextBaseline) {
+//          AmountText(
+//            amount: entry.available,
+//            wholePartFont: .system(
+//              size: UIFont.preferredFont(
+//                forTextStyle: .largeTitle).pointSize * 2)
+//          )
+//          .minimumScaleFactor(0.1)
+//          .bold()
+//          .foregroundStyle(.green)
+//          
+//          Text("")
+//          
+//          AmountText(
+//            amount: entry.spent,
+//            wholePartFont: .system(
+//              size: UIFont.preferredFont(
+//                forTextStyle: .largeTitle).pointSize)
+//          )
+//          .minimumScaleFactor(0.1)
+//          .bold()
+//        }
+//        
+//        GridRow(alignment: .lastTextBaseline) {
+//          Text("Available today")
+//            .fontWeight(.light)
+//          
+//          Text("")
+//          
+//          Text("Spent")
+//            .fontWeight(.light)
+//        }
+//      }
+//    }
+//  }
+//}
+
+// MARK: Entry -
+
+// MARK: Provider -
+
+// MARK: Double extension -
+// TODO: Refactor
+
+extension Double {
+  var wholePart: Int {
+    Int(self)
+  }
+  
+  var fractionPartInt: Int {
+    // Take first two decimal digits
+    abs(Int(self.truncatingRemainder(dividingBy: 1) * 100))
+  }
+  
+  var fractionPart: String {
+    String(
+      // Show 2 decimal places
+      format: "%02d",
+      fractionPartInt
+    )
+  }
+}
+
+// MARK: Amount label -
+// TODO: Refactor if not too different from main?
```

Comment:
TODO: Clean up 

- See if the changes here (content animation) can be used in the view in the main app?
- If yes, use that view in the widget